app/controllers/products_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ProductsController < ApplicationController
+  def new
+    @product = Product.new 
+  end
+
+  def show 
+    @product = Product.find(params[:id])
+  end
+
+  def create 
+    @product = Product.new(product_params)
+    if @product.save
+        redirect_to @product 
+    else 
+        render :new 
+    end
+  end
+  
+  private 
+
+  def product_params
+    params.require(:product).permit(:title, :description, :price, :image)
+  end
+end
```

Comment:
We need to have a new line at the end of the (every)file, even github complains about it here if you see the red circle with the line.

more info about this here https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline#:~:text=So%2C%20the%20reasons%20are%3A,t%20end%20with%20a%20newline.